src/pymovements/synthetic/step_function.py

```diff
@@ -31,6 +31,7 @@ def step_function(
         values: list[float | tuple[float, ...]],
         start_value: float | tuple[float, ...] = 0,
         noise: float = 0,
+        dtype: np.dtype = np.float32,
```

Comment:
I just confirmed that using [`numpy.typing.DTypeLike`](https://numpy.org/devdocs/reference/typing.html#numpy.typing.DTypeLike) will get rid of the mypy error, so please adjust the type accordingly.